code/modules/mining/ores_coins.dm

```diff
@@ -287,7 +287,7 @@
 		flick("coin_[cmineral]_flip", src)
 		icon_state = "coin_[cmineral]_[coinflip]"
 		playsound(user.loc, 'sound/items/coinflip.ogg', 50, 1)
-		if(do_after(user, 15))
+		if(do_after(user, 15, target = src))

```

Comment:
This will cause problems, especially if you throw the coin while it's flipping.

Change it to this.

```
var/oldloc = loc
sleep(15)
if(loc == oldloc && user && !user.incapacitated())
```